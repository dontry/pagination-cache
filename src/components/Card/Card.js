import React from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core/styles";
import MuiCard from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";

const useStyles = makeStyles({
  card: {
    minWidth: 300
  },
  title: {
    fontSize: 14
  }
});

const Card = ({ data = {}, openCard }) => {
  const { card, title } = useStyles();
  const { id, state, number, application, shortDescription, assignee } = data;

  const handleClick = () => {
    openCard(id);
  };
  return (
    <MuiCard className={card}>
      <CardContent>
        <Typography variant="h5">{state}</Typography>
        <Typography className={title} gutterBottom>
          {number}
        </Typography>
        <Typography variant="h4">Application: {application}</Typography>
        <Typography variant="h4">Assignee: {assignee}</Typography>
        <Typography variant="body2">{shortDescription}</Typography>
      </CardContent>
      <CardActions>
        <Button data-testid="learn-more" size="small" onClick={handleClick}>
          Learn more
        </Button>
      </CardActions>
    </MuiCard>
  );
};

export const CoreDataType = {
  id: PropTypes.string,
  state: PropTypes.string,
  number: PropTypes.string,
  application: PropTypes.string,
  shortDescription: PropTypes.string,
  assignee: PropTypes.string
};

Card.propTypes = {
  data: PropTypes.shape(CoreDataType),
  openCard: PropTypes.func.isRequired
};

export default Card;
