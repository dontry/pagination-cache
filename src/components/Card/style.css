.card {
  display: flex;
  min-width: 200px;
  flex-direction: column;
  align-items: flex-start;
  text-align: left;
}

.card-subheading {
  color: #acacac;
}

.learnmore .MuiButton-label {
  font-weight: 600;
}
