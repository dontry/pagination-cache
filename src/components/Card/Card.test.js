import React from "react";
import  {render,fireEvent } from "@testing-library/react"
import Card from './Card'

const data = {
 "id": "",
  "number": "INCXXXXX",
  "lastUpdateDate": "YYYY-MM-DD hh:mm:ss",
  "type": "INCIDENT",
  "state": "New",
  "shortDescription": "description",
  "application": "System",
  "assignee": "John Doe",
}

describe('Card Component', () => {
  it("renders Card with data", () => {
    const openCard = jest.fn();
    const {container,  queryByText} = render(<Card data={data} openCard={openCard}/>)
    expect(queryByText(data.number)).toBeTruthy()
    expect(queryByText(data.state)).toBeTruthy();
    expect(queryByText(data.shortDescription)).toBeTruthy();
    expect(queryByText(`Assignee: ${data.assignee}`)).toBeTruthy();
    expect(queryByText(`Application: ${data.application}`)).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  })

  it("should fire openCard function when 'Learn more' button is pressed", () => {
    const openCard = jest.fn();
    const {getByText} = render(<Card data={data} openCard={openCard}/>)
    fireEvent.click(getByText("Learn more"))
    expect(openCard).toBeCalled();
  })
})
