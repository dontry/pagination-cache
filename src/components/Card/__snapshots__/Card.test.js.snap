// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders Card with core data 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation1 MuiCard-root card MuiPaper-rounded"
>
  <div
    class="MuiCardContent-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      New
    </p>
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      INCXXXXX
    </h5>
    <p
      class="MuiTypography-root card-subheading MuiTypography-body1"
    >
      Application: 
      System
    </p>
    <p
      class="MuiTypography-root card-subheading MuiTypography-body1 MuiTypography-gutterBottom"
    >
      Assignee: 
      John Doe
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2"
    >
      description
    </p>
  </div>
  <div
    class="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      class="MuiButtonBase-root MuiButton-root learnmore MuiButton-text MuiButton-sizeSmall"
      data-testid="learn-more"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Learn more
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
